### ISSUE TITLE
Custom operator definitions might result in wrong warnings

### ISSUE THREAD

**Issue by evnu** [rrrene/credo#454]
When defining a custom operator, Credo tries to apply the custom operator to the arguments provided, which fails for `@spec` and the function clause.

### Environment

* Credo version (`mix credo -v`): 0.8.7
* Erlang/Elixir version (`elixir -v`): Elixir 1.5.1
* Operating system: ArchLinux

### What were you trying to do?
Define a custom operator  with an accompanying typespec:

```elixir
defmodule Red do
  @moduledoc false
  @spec number - number :: number
  def a - a do
    a + 1
  end
end
```

### Expected outcome

No output from credo.

### Actual outcome

```
  Warnings - please take a look                                                 
┃ 
┃ [W] ↗ Operation will always return 0.
┃       lib/red.ex:5 #(Red.-)
┃ [W] ↗ Operation will always return 0.
┃       lib/red.ex:4 #(Red)
```

**Comment by evnu** [comment]
Of course, using `def a - a` might be considered to be non-sense. The following example exhibits the warning only for the `@spec`:

```elixir
defmodule Red do
  @moduledoc false

  @spec number - number :: number
  def a - _b do
    a + 1
  end
end
```

Output:

```
  Warnings - please take a look                                                 
┃ 
┃ [W] ↗ Operation will always return 0.
┃       lib/red.ex:4 #(Red)
```

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by evnu** [comment]
Works like a charm, thank you!
